<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Overview of the condiments workflow • condiments</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Overview of the condiments workflow">
<meta property="og:description" content="condiments">
<meta property="og:image" content="http://hectorrdb.github.io//condiments/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">condiments</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.8.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/condiments.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/controls.html">More controls for the tests used in the condiments workflow</a>
    </li>
    <li>
      <a href="../articles/examples.html">Generating toy datasets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HectorRDB/condiments">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="condiments_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Overview of the condiments workflow</h1>
                        <h4 class="author">Hector Roux de Bézieux</h4>
            
            <h4 class="date">09 March , 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HectorRDB/condiments/blob/master/vignettes/condiments.Rmd"><code>vignettes/condiments.Rmd</code></a></small>
      <div class="hidden name"><code>condiments.Rmd</code></div>

    </div>

    
    
<div id="initial-pre-processing" class="section level1">
<h1 class="hasAnchor">
<a href="#initial-pre-processing" class="anchor"></a>Initial pre-processing</h1>
<div id="generating-a-synthetic-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#generating-a-synthetic-dataset" class="anchor"></a>Generating a synthetic dataset</h2>
<p>We will use a synthetic dataset to illustrate the functionalities of the <em>condiments</em> package. We start directly with a dataset where the following steps are assumed to have been run:</p>
<ul>
<li>Obtaining count matrices for each setting (i.e. each condition).</li>
<li>Integration and normalization between the conditions.</li>
<li>Reduced Dimension Estimations</li>
<li>(Clustering)</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># For analysis</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://hectorrdb.github.io/condiments/index.html">condiments</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">slingshot</span><span class="op">)</span>
<span class="co"># For data manipulation</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="co"># For visualization</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sjmgarnier/viridis">viridis</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2071</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"toy_dataset"</span>, package <span class="op">=</span> <span class="st">"condiments"</span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="va">toy_dataset</span><span class="op">$</span><span class="va">sd</span></code></pre></div>
<p>As such, we start with a matrix <code>df</code> of metadata for the cells: coordinates in a reduced dimension space <code>(Dim1, Dim2)</code>, a vector of conditions assignments <code>conditions</code> (A or B) and a lineage assignment.</p>
</div>
<div id="vizualisation" class="section level2">
<h2 class="hasAnchor">
<a href="#vizualisation" class="anchor"></a>Vizualisation</h2>
<p>We can first plot the cells on the reduced dimensions</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-3-1.png" width="700" style="display: block; margin: auto;"></p>
<p>We can also visualize the underlying skeleton structure of the two conditions.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">toy_dataset</span><span class="op">$</span><span class="va">mst</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_path</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">toy_dataset</span><span class="op">$</span><span class="va">mst</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">lineages</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">conditions</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>col <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="differential-topology" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-topology" class="anchor"></a>Differential Topology</h1>
<div id="exploratory-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#exploratory-analysis" class="anchor"></a>Exploratory analysis</h2>
<p>We can then compute the <strong>imbalance score</strong> of each cell using the <em>imbalance_score</em> function.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imbalance_score.html">imbalance_score</a></span><span class="op">(</span>Object <span class="op">=</span> <span class="va">df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">Dim1</span>, <span class="va">Dim2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="op">)</span>,
                          conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span><span class="op">)</span>
<span class="va">df</span><span class="op">$</span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">scores</span><span class="op">$</span><span class="va">scores</span>
<span class="va">df</span><span class="op">$</span><span class="va">scaled_scores</span> <span class="op">&lt;-</span> <span class="va">scores</span><span class="op">$</span><span class="va">scaled_scores</span></code></pre></div>
<p>There are two types of scores. The raw score is computed on each cell and looks at the condition distribution of its neighbors compared the the overall distribution. The size of the neighborhood can be set using the <code>k</code> argument, which specify the number of neighbors to consider. Higher values means more local imbalance.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-6-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The local scores are quite noisy so we can then use local smoothers to smooth the scores of individual cells. The smoothness is dictated by the <code>smooth</code> argument. Those smoothed scores were also computed using the <code>imbalance_score</code> function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">scaled_scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_color_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-7-1.png" width="700" style="display: block; margin: auto;"></p>
<p>As could be guessed from the original plot, the bottom lineage shows a lot of imbalance while the top one does not. The imbalance score can be used to check: + If the integration has been successful. In general, some regions should be balanced + To identify the regions of imbalance for further analyses.</p>
</div>
<div id="trajectory-inference" class="section level2">
<h2 class="hasAnchor">
<a href="#trajectory-inference" class="anchor"></a>Trajectory Inference</h2>
<p>The first step of our workflow is to decide whether or not to infer the trajectories separately or not. On average, it is better to infer a common trajectory, since a) this allow for a wider range of downstream analyses, and b) more cells are used to estimate the trajectory. However, the condition effect might be strong enough to massively disrupt the differentiation process, which would require fitting separate trajectories.</p>
<p><strong>slingshot</strong><span class="citation">(Street et al. 2018)</span> relies on a reduced dimensionality reduction representation of the data, as well as on cluster labels. We can visualize those below:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">cl</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-8-1.png" width="700" style="display: block; margin: auto;"> The <strong>topologyTest</strong> assess the quality of the common trajectory inference done by slingshot and test whether we should fit a common or separate trajectory. This test relies on repeated permutations of the conditions followed by trajectory inference so it can take a few seconds.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="va">df</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Dim1"</span>, <span class="st">"Dim2"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">sds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingshot.html">slingshot</a></span><span class="op">(</span><span class="va">rd</span>, <span class="va">df</span><span class="op">$</span><span class="va">cl</span><span class="op">)</span></code></pre></div>
<pre><code>## Using full covariance matrix</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Takes ~1m30s to run</span>
<span class="va">top_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/topologyTest.html">topologyTest</a></span><span class="op">(</span>sds <span class="op">=</span> <span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span><span class="op">)</span></code></pre></div>
<pre><code>## Generating permuted trajectories</code></pre>
<pre><code>## Running KS-mean test</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">top_res</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">method</th>
<th align="left">thresh</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">KS_mean</td>
<td align="left">0.01</td>
<td align="right">0</td>
<td align="right">1</td>
</tr></tbody>
</table>
<p>The test clearly fails to reject the null that we can fit a common trajectory so we can continue with the <code>sds</code> object. This will facilitate downstream analysis. For an example of how to proceed if the <strong>topologyTest</strong> reject the null, we invite the user to refer to <a href="https://hectorrdb.github.io/condimentsPaper/articles/KRAS.html">relevant case study used in our paper</a>.</p>
<p>We can thus visualize the trajectory</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">cl</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_path</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingCurves.html">slingCurves</a></span><span class="op">(</span><span class="va">sds</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">s</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingCurves.html">slingCurves</a></span><span class="op">(</span><span class="va">sds</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">ord</span>, <span class="op">]</span> <span class="op">%&gt;%</span>
              <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_path</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingCurves.html">slingCurves</a></span><span class="op">(</span><span class="va">sds</span><span class="op">)</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">s</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingCurves.html">slingCurves</a></span><span class="op">(</span><span class="va">sds</span><span class="op">)</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">ord</span>, <span class="op">]</span> <span class="op">%&gt;%</span>
              <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>,
            col <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="cn">NULL</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-10-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="differential-progression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-progression" class="anchor"></a>Differential Progression</h1>
<p>Even though we can fit a common trajectory, it does not mean that the cells will differentiate similarly between the conditions. The first question we can ask is: for a given lineage, are cells equally represented along pseudotime between conditions?</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">psts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html">slingPseudotime</a></span><span class="op">(</span><span class="va">sds</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>cells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>,
         conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html">starts_with</a></span><span class="op">(</span><span class="st">"curve"</span><span class="op">)</span>, values_to <span class="op">=</span> <span class="st">"pseudotime"</span>, names_to <span class="op">=</span> <span class="st">"lineages"</span><span class="op">)</span></code></pre></div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">psts</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pseudotime</span>, fill <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">lineages</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-12-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The pseudotime distributions are identical across conditions for the first lineage but there are clear differences between the two conditions in the second lineage.</p>
</div>
<div id="testing-for-differential-progression" class="section level2">
<h2 class="hasAnchor">
<a href="#testing-for-differential-progression" class="anchor"></a>Testing for differential progression</h2>
<p>To test for differential progression, we use the <strong>progressionTest</strong>. The test can be run with <code>global = TRUE</code> to test when pooling all lineages, or <code>lineages = TRUE</code> to test every lineage independently, or both. Several tests are implemented in the <strong>progressionTest</strong>. function. Here, we will use the default, the custom KS test <span class="citation">(Smirnov 1939)</span>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prog_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/progressionTest.html">progressionTest</a></span><span class="op">(</span><span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span>, global <span class="op">=</span> <span class="cn">TRUE</span>, lineages <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">prog_res</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">lineage</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">All</td>
<td align="right">5.5068236</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="right">0.0344087</td>
<td align="right">0.9940214</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="right">0.4699368</td>
<td align="right">0.0000000</td>
</tr>
</tbody>
</table>
<p>As expected, there is a global difference over all lineages, which is driven by differences of distribution across lineage 2 (i.e. the bottom one).</p>
</div>
</div>
<div id="differential-differentiation" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-differentiation" class="anchor"></a>Differential differentiation</h1>
<p>Even though we can fit a common trajectory, it does not mean that the cells will differentiate similarly between the conditions. The first question we can ask is: for a given lineage, are cells equally between the two lineages for the two conditions?</p>
<div id="vizualisation-1" class="section level2">
<h2 class="hasAnchor">
<a href="#vizualisation-1" class="anchor"></a>Vizualisation</h2>
<p>Visualizing differences 2D distributions can be somewhat tricky. However, it is important to note that the sum of all lineage weights should sum to 1. As such, we can only plot the weights for the first lineage.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span><span class="op">$</span><span class="va">weight_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html">slingCurveWeights</a></span><span class="op">(</span><span class="va">sds</span>, as.probs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">weight_1</span>, fill <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Curve weight for the first lineage"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-14-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The distribution has tri modes, which is very often the case for two lineages: + A weight around 0 represent a cell that is mostly assigned to the other lineage (i.e. lineage 2 here) + A weight around .5 represent a cell that is equally assigned to both lineages, i.e. before the bifurcation. + A weight around 1 represent a cell that is mostly assigned to this lineage (i.e. lineage 1 here)</p>
<p>In condition A, we have many more cells with a weight of 0 and, since those are density plots, fewer cells with weights around .5 and 1. Visually, we can guess that cells in condition B differentiate preferentially along lineage 1.</p>
</div>
<div id="testing-for-differential-differentiation" class="section level2">
<h2 class="hasAnchor">
<a href="#testing-for-differential-differentiation" class="anchor"></a>Testing for differential differentiation</h2>
<p>To test for differential differentiation, we use the <strong>differentiationTest</strong>. The test can be run with <code>global = TRUE</code> to test when pooling all pairs of lineages, or <code>pairwise = TRUE</code> to test every pair independently, or both. Here, there is only one pair so the options are equivalent. Several tests are implemented in the <strong>differentiationTest</strong>. function. Here, we will use the default, the classifier test<span class="citation">(Lopez-Paz and Oquab 2016)</span>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span>
<span class="va">dif_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/differentiationTest.html">differentiationTest</a></span><span class="op">(</span><span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span>, global <span class="op">=</span> <span class="cn">FALSE</span>, pairwise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## note: only 1 unique complexity parameters in default grid. Truncating the grid to 1 .</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## note: only 1 unique complexity parameters in default grid. Truncating the grid to 1 .</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">dif_res</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">pair</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">1vs2</td>
<td align="right">0.6476577</td>
<td align="right">5.6e-06</td>
</tr></tbody>
</table>
<p>As could be guessed from the plot, we have clear differential differentiation.</p>
</div>
</div>
<div id="differential-expression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential Expression</h1>
<p>The workflow above focus on global differences, looking at broad patterns of differentiation. While this is a necessary first step, gene-level information is also quite meaningful.</p>
<p>To do so requires <strong>tradeSeq</strong><span class="citation">(Van den Berge et al. 2020)</span> &gt; 1.3.0. Considering that we have a count matrix <code>counts</code>, the basic workflow is:s</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">tradeSeq</span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">fitGAM</span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span>, sds <span class="op">=</span> <span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span><span class="op">)</span>
<span class="va">cond_genes</span> <span class="op">&lt;-</span> <span class="fu">conditionTest</span><span class="op">(</span><span class="va">sds</span><span class="op">)</span></code></pre></div>
<p>For more details on fitting the smoothers, we refer users to <a href="http://statomics.github.io/tradeSeq">the tradeSeq website</a> and to the accompanying <a href="https://kstreet13.github.io/bioc2020trajectories/articles/workshopTrajectories.html#differential-expression-1">Bioconductor workshop</a>.</p>
</div>
<div id="conclusion" class="section level1">
<h1 class="hasAnchor">
<a href="#conclusion" class="anchor"></a>Conclusion</h1>
<p>For both of the above procedures, it is important to note that we are making multiple comparisons (in this case, 5). The p-values we obtain from these tests should be corrected for multiple testing, especially for trajectories with a large number of lineages.</p>
<p>That said, trajectory inference is often one of the last computational methods in a very long analysis pipeline (generally including gene-level quantification, gene filtering / feature selection, and dimensionality reduction). Hence, we strongly discourage the reader from putting too much faith in any p-value that comes out of this analysis. Such values may be useful suggestions, indicating particular features or cells for follow-up study, but should generally not be treated as meaningful statistical quantities.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] caret_6.0-86       lattice_0.20-41    viridis_0.5.1      viridisLite_0.3.0 
##  [5] RColorBrewer_1.1-2 ggplot2_3.3.3      tidyr_1.1.3        dplyr_1.0.5       
##  [9] slingshot_1.8.0    princurve_2.1.6    condiments_0.8.0   knitr_1.31        
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-0            deldir_0.2-10              
##   [3] ellipsis_0.3.1              class_7.3-17               
##   [5] rprojroot_2.0.2             XVector_0.30.0             
##   [7] GenomicRanges_1.42.0        fs_1.5.0                   
##   [9] spatstat.data_2.0-0         farver_2.1.0               
##  [11] prodlim_2019.11.13          lubridate_1.7.10           
##  [13] codetools_0.2-18            splines_4.0.3              
##  [15] polyclip_1.10-0             pROC_1.17.0.1              
##  [17] kernlab_0.9-29              compiler_4.0.3             
##  [19] Matrix_1.3-0                htmltools_0.5.0            
##  [21] tools_4.0.3                 igraph_1.2.6               
##  [23] gtable_0.3.0                glue_1.4.2                 
##  [25] GenomeInfoDbData_1.2.4      RANN_2.6.1                 
##  [27] reshape2_1.4.4              Rcpp_1.0.5                 
##  [29] spatstat_1.64-1             Biobase_2.50.0             
##  [31] pkgdown_1.6.1               vctrs_0.3.6                
##  [33] ape_5.4-1                   nlme_3.1-151               
##  [35] iterators_1.0.13            timeDate_3043.102          
##  [37] gower_0.2.2                 xfun_0.19                  
##  [39] stringr_1.4.0               Ecume_0.9.0                
##  [41] lifecycle_1.0.0             goftest_1.2-2              
##  [43] zlibbioc_1.36.0             MASS_7.3-53                
##  [45] scales_1.1.1                ipred_0.9-10               
##  [47] ragg_1.1.1                  MatrixGenerics_1.2.1       
##  [49] spatstat.utils_2.0-0        parallel_4.0.3             
##  [51] SummarizedExperiment_1.20.0 SingleCellExperiment_1.12.0
##  [53] yaml_2.2.1                  memoise_1.1.0              
##  [55] pbapply_1.4-3               gridExtra_2.3              
##  [57] rpart_4.1-15                stringi_1.5.3              
##  [59] highr_0.8                   S4Vectors_0.28.1           
##  [61] desc_1.3.0                  randomForest_4.6-14        
##  [63] foreach_1.5.1               e1071_1.7-4                
##  [65] BiocGenerics_0.36.0         BiocParallel_1.24.1        
##  [67] lava_1.6.8.1                GenomeInfoDb_1.26.2        
##  [69] rlang_0.4.10                pkgconfig_2.0.3            
##  [71] systemfonts_1.0.1           matrixStats_0.58.0         
##  [73] bitops_1.0-6                evaluate_0.14              
##  [75] purrr_0.3.4                 tensor_1.5                 
##  [77] recipes_0.1.15              labeling_0.4.2             
##  [79] transport_0.12-2            tidyselect_1.1.0           
##  [81] plyr_1.8.6                  magrittr_2.0.1             
##  [83] R6_2.5.0                    IRanges_2.24.1             
##  [85] generics_0.1.0              DelayedArray_0.16.2        
##  [87] pillar_1.4.7                withr_2.3.0                
##  [89] mgcv_1.8-33                 survival_3.2-7             
##  [91] abind_1.4-5                 RCurl_1.98-1.2             
##  [93] nnet_7.3-14                 tibble_3.0.4               
##  [95] crayon_1.4.1                rmarkdown_2.7              
##  [97] grid_4.0.3                  data.table_1.14.0          
##  [99] ModelMetrics_1.2.2.2        digest_0.6.27              
## [101] textshaping_0.3.1           stats4_4.0.3               
## [103] munsell_0.5.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Lopez-Paz2016">
<p>Lopez-Paz, David, and Maxime Oquab. 2016. “Revisiting Classifier Two-Sample Tests.” <em>Arxiv</em>, October, 1–15. <a href="http://arxiv.org/abs/1610.06545">http://arxiv.org/abs/1610.06545</a>.</p>
</div>
<div id="ref-smirnov1939estimation">
<p>Smirnov, Nikolai V. 1939. “On the Estimation of the Discrepancy Between Empirical Curves of Distribution for Two Independent Samples.” <em>Bull. Math. Univ. Moscou</em> 2 (2): 3–14.</p>
</div>
<div id="ref-Street2018a">
<p>Street, Kelly, Davide Risso, Russell B. Fletcher, Diya Das, John Ngai, Nir Yosef, Elizabeth Purdom, and Sandrine Dudoit. 2018. “Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics.” <em>BMC Genomics</em> 19 (1): 477. <a href="https://doi.org/10.1186/s12864-018-4772-0">https://doi.org/10.1186/s12864-018-4772-0</a>.</p>
</div>
<div id="ref-VandenBerge2020">
<p>Van den Berge, Koen, Hector Roux de Bézieux, Kelly Street, Wouter Saelens, Robrecht Cannoodt, Yvan Saeys, Sandrine Dudoit, and Lieven Clement. 2020. “Trajectory-based differential expression analysis for single-cell sequencing data.” <em>Nature Communications</em> 11 (1): 1201. <a href="https://doi.org/10.1038/s41467-020-14766-3">https://doi.org/10.1038/s41467-020-14766-3</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Hector Roux de Bezieux, Koen Van den Berge, Kelly Street.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
