<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>condiments • condiments</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="condiments">
<meta property="og:description" content="condiments">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">condiments</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/condiments.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HectorRDB/condiments/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="condiments_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>condiments</h1>
                        <h4 data-toc-skip class="author">Hector Roux de Bézieux</h4>
            
            <h4 data-toc-skip class="date">26 February , 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HectorRDB/condiments/blob/master/vignettes/condiments.Rmd" class="external-link"><code>vignettes/condiments.Rmd</code></a></small>
      <div class="hidden name"><code>condiments.Rmd</code></div>

    </div>

    
    
<div id="initial-pre-processing" class="section level1">
<h1 class="hasAnchor">
<a href="#initial-pre-processing" class="anchor"></a>Initial pre-processing</h1>
<div id="generating-a-synthetic-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#generating-a-synthetic-dataset" class="anchor"></a>Generating a synthetic dataset</h2>
<p>We will use a synthetic dataset to illustrate the functionalities of the <em>condiments</em> package. We start directly with a dataset where the following steps are assumed to have been run:</p>
<ul>
<li>Obtaining count matrices for each setting (i.e. each condition).</li>
<li>Integration and normalization between the conditions.</li>
<li>Reduced Dimension Estimations</li>
<li>(Clustering)</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hectorrdb.github.io/condiments/index.html" class="external-link">condiments</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">slingshot</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required package: princurve</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sjmgarnier/viridis" class="external-link">viridis</a></span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required package: viridisLite</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2071</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"toy_dataset"</span>, package <span class="op">=</span> <span class="st">"condiments"</span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="va">toy_dataset</span><span class="op">$</span><span class="va">sd</span></code></pre></div>
<p>As such, we start with a matrix <code>df</code> of metadata the cells: coordinates in a reduced dimension space <code>(Dim1, Dim2)</code>, a vector of conditions assignments <code>conditions</code> (A or B) and a lineage assignment.</p>
</div>
<div id="vizualisation" class="section level2">
<h2 class="hasAnchor">
<a href="#vizualisation" class="anchor"></a>Vizualisation</h2>
<p>We can first plot the cells on the reduced dimensions</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-3-1.png" width="700" style="display: block; margin: auto;"></p>
<p>We can also visualize the underlying skeleton structure of the two conditions.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">toy_dataset</span><span class="op">$</span><span class="va">mst</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="co"># We do two groups here to represent the two paths in the tree structure</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_path</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">toy_dataset</span><span class="op">$</span><span class="va">mst</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Dim2</span> <span class="op">&gt;=</span> <span class="fl">0</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_path</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">toy_dataset</span><span class="op">$</span><span class="va">mst</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Dim2</span> <span class="op">&lt;=</span> <span class="fl">0</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">conditions</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="differential-topology" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-topology" class="anchor"></a>Differential Topology</h1>
<div id="exploratory-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#exploratory-analysis" class="anchor"></a>Exploratory analysis</h2>
<p>We can then compute the <strong>imbalance score</strong> of each cell using the <em>imbalance_score</em> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imbalance_score.html">imbalance_score</a></span><span class="op">(</span>Object <span class="op">=</span> <span class="va">df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Dim1</span>, <span class="va">Dim2</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>,
                          conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span><span class="op">)</span>
<span class="va">df</span><span class="op">$</span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">scores</span><span class="op">$</span><span class="va">scores</span>
<span class="va">df</span><span class="op">$</span><span class="va">scaled_scores</span> <span class="op">&lt;-</span> <span class="va">scores</span><span class="op">$</span><span class="va">scaled_scores</span></code></pre></div>
<p>There are two types of scores. The raw score is computed on each cell and looks at the condition distribution of its neighbors compared the the overall distribution. The size of the neighborhood can be set using the <code>k</code> argument, which specify the number of neighbors to consider. Higher values means more local imbalance.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-6-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The local scores are quite noisy so we can then use local smoothers to smooth the scores of individual cells. The smoothness is dictated by the <code>smooth</code> argument.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">scaled_scores</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"C"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-7-1.png" width="700" style="display: block; margin: auto;"></p>
<p>As could be guessed from the original plot, the bottom lineage shows a lot of imbalance while the top one does not. The imbalance score can be used to check whether the integration has been successful (at least some regions should be balanced), or to identify the regions of imbalance for further analyses.</p>
</div>
<div id="trajectory-inference" class="section level2">
<h2 class="hasAnchor">
<a href="#trajectory-inference" class="anchor"></a>Trajectory Inference</h2>
<p>The first step of our workflow is to decide whether or not to infer the trajectories separately or not. On average, it is better to infer a common trajectory, since a) this allow for a wider range of downstream analyses, and b) more cells are used to estimate the trajectory. However, the condition effect might be strong enough to massively disrupt the differentiation process, which would require fitting separate trajectories.</p>
<p>The <strong>topologyTest</strong> assess the quality of the common trajectory inference done by slingshot and test whether we should fit a common or separate trajectory. This test relies on repeated permutations of the conditions followed by trajectory inference so it can take a few seconds.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dim1</span>, y <span class="op">=</span> <span class="va">Dim2</span>, col <span class="op">=</span> <span class="va">cl</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-8-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingshot.html" class="external-link">slingshot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">df</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="va">df</span><span class="op">$</span><span class="va">cl</span><span class="op">)</span></code></pre></div>
<pre><code>## Using full covariance matrix</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Takes ~1mn30s to run</span>
<span class="fu"><a href="../reference/topologyTest.html">topologyTest</a></span><span class="op">(</span>sds <span class="op">=</span> <span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span><span class="op">)</span></code></pre></div>
<pre><code>## Generating permuted trajectories</code></pre>
<pre><code>## Running KS-mean test</code></pre>
<pre><code>##    method thresh statistic p.value
## 1 KS_mean   0.01         0       1</code></pre>
<p>The test fails to reject the null that we can fit a common trajectory so we can continue with the <code>sds</code> object.</p>
</div>
</div>
<div id="differential-progression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-progression" class="anchor"></a>Differential Progression</h1>
<p>Even though we can fit a common trajectory, it does not mean that the cells will differentiate similarly between the conditions. The first question we can ask is: for a given lineage, are cells equally represented along pseudotime between conditions?</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span><span class="op">$</span><span class="va">pst1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html" class="external-link">slingPseudotime</a></span><span class="op">(</span><span class="va">sds</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>
<span class="va">df</span><span class="op">$</span><span class="va">pst2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/slingshot/man/slingPseudotime.html" class="external-link">slingPseudotime</a></span><span class="op">(</span><span class="va">sds</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span></code></pre></div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pst1</span>, fill <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Pseudotime for lineage 1"</span>, title <span class="op">=</span> <span class="st">"Pseudotime distribution for lineage 1"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-11-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The pseudotime distributions are identical across lineages.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pst2</span>, fill <span class="op">=</span> <span class="va">conditions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Pseudotime for lineage 2"</span>, title <span class="op">=</span> <span class="st">"Pseudotime distribution for lineage 2"</span><span class="op">)</span></code></pre></div>
<p><img src="condiments_files/figure-html/unnamed-chunk-12-1.png" width="700" style="display: block; margin: auto;"></p>
<p>There are clear differences between the two conditions</p>
</div>
<div id="testing-for-differential-progression" class="section level2">
<h2 class="hasAnchor">
<a href="#testing-for-differential-progression" class="anchor"></a>Testing for differential progression</h2>
<p>To test for differential progression, we use the <strong>progressionTest</strong>. The test can be run with <code>global = TRUE</code> to test when pooling all lineages, or <code>lineages = TRUE</code> to test every lineage independently, or both. Several tests are implemented in the <strong>progressionTest</strong>. function. Here, we will use the default, the custom KS test.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/progressionTest.html">progressionTest</a></span><span class="op">(</span><span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span>, global <span class="op">=</span> <span class="cn">TRUE</span>, lineages <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>##   lineage  statistic      p.value
## 1     All 5.50682358 1.826830e-08
## 2       1 0.03440871 9.940214e-01
## 3       2 0.46993683 2.200000e-16</code></pre>
<p>As expected, there is a global difference over all lineages, which is driven by differences of distribution across lineage 2 (i.e. the bottom one).</p>
</div>
<div id="testing-for-differential-differentiation" class="section level2">
<h2 class="hasAnchor">
<a href="#testing-for-differential-differentiation" class="anchor"></a>Testing for differential differentiation</h2>
<p>Another way to look at this is to consider how the cells of one condition differentiate between two lineages, compared to other conditions.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span>
<span class="fu"><a href="../reference/differentiationTest.html">differentiationTest</a></span><span class="op">(</span><span class="va">sds</span>, conditions <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">conditions</span>, global <span class="op">=</span> <span class="cn">FALSE</span>, pairwise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## note: only 1 unique complexity parameters in default grid. Truncating the grid to 1 .</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## note: only 1 unique complexity parameters in default grid. Truncating the grid to 1 .</code></pre>
<pre><code>## Registered S3 method overwritten by 'cli':
##   method     from    
##   print.boxx spatstat</code></pre>
<pre><code>## # A tibble: 1 x 3
##   pair  statistic    p.value
##   &lt;chr&gt;     &lt;dbl&gt;      &lt;dbl&gt;
## 1 1vs2      0.648 0.00000559</code></pre>
</div>
</div>
<div id="differential-expression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential Expression</h1>
<p>The workflow above focus on global differences, looking at broad patterns of differentiation. While this is a necessary first step, gene-level information is also quite meaningful. To do so, we refer the reader to versions of <strong>tradeSeq</strong> &gt; 1.3.0, and to the accompanying <a href="https://kstreet13.github.io/bioc2020trajectories/articles/workshopTrajectories.html#differential-expression-1" class="external-link">Bioconductor workshop</a>.</p>
</div>
<div id="conclusion" class="section level1">
<h1 class="hasAnchor">
<a href="#conclusion" class="anchor"></a>Conclusion</h1>
<p>An earlier version of this vignette has been produced by Kelly Street and Koen Van den Berge and can be found <a href="https://bioconductor.org/packages/release/bioc/vignettes/slingshot/inst/doc/conditionsVignette.html" class="external-link">here</a>.</p>
<p>For both of the above procedures, it is important to note that we are making multiple comparisons (in this case, 2). The p-values we obtain from these tests should be corrected for multiple testing, especially for trajectories with a large number of lineages.</p>
<p>That said, trajectory inference is often one of the last computational methods in a very long analysis pipeline (generally including gene-level quantification, gene filtering / feature selection, and dimensionality reduction). Hence, we strongly discourage the reader from putting too much faith in any p-value that comes out of this analysis. Such values may be useful suggestions, indicating particular features or cells for follow-up study, but should not be treated as meaningful statistical quantities.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.4 (2021-02-15)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] caret_6.0-86       lattice_0.20-41    ggplot2_3.3.3      viridis_0.5.1     
##  [5] viridisLite_0.3.0  dplyr_1.0.4        RColorBrewer_1.1-2 slingshot_1.8.0   
##  [9] princurve_2.1.6    condiments_0.1.0   knitr_1.31        
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-0            deldir_0.2-10              
##   [3] ellipsis_0.3.1              class_7.3-18               
##   [5] rprojroot_2.0.2             XVector_0.30.0             
##   [7] GenomicRanges_1.42.0        fs_1.5.0                   
##   [9] rstudioapi_0.13             spatstat.data_2.0-0        
##  [11] farver_2.0.3                prodlim_2019.11.13         
##  [13] fansi_0.4.2                 lubridate_1.7.10           
##  [15] codetools_0.2-18            splines_4.0.4              
##  [17] cachem_1.0.4                polyclip_1.10-0            
##  [19] pROC_1.17.0.1               kernlab_0.9-29             
##  [21] compiler_4.0.4              assertthat_0.2.1           
##  [23] Matrix_1.3-2                fastmap_1.1.0              
##  [25] cli_2.3.1                   htmltools_0.5.1.1          
##  [27] tools_4.0.4                 igraph_1.2.6               
##  [29] gtable_0.3.0                glue_1.4.2                 
##  [31] GenomeInfoDbData_1.2.4      RANN_2.6.1                 
##  [33] reshape2_1.4.4              Rcpp_1.0.6                 
##  [35] spatstat_1.64-1             Biobase_2.50.0             
##  [37] pkgdown_1.6.1.9000          vctrs_0.3.6                
##  [39] ape_5.4-1                   nlme_3.1-152               
##  [41] iterators_1.0.13            timeDate_3043.102          
##  [43] gower_0.2.2                 xfun_0.21                  
##  [45] stringr_1.4.0               ps_1.5.0                   
##  [47] Ecume_0.9.0                 lifecycle_1.0.0            
##  [49] goftest_1.2-2               zlibbioc_1.36.0            
##  [51] MASS_7.3-53.1               scales_1.1.1               
##  [53] ipred_0.9-9                 ragg_1.1.1                 
##  [55] MatrixGenerics_1.2.1        spatstat.utils_2.0-0       
##  [57] parallel_4.0.4              SummarizedExperiment_1.20.0
##  [59] SingleCellExperiment_1.12.0 yaml_2.2.1                 
##  [61] memoise_2.0.0               pbapply_1.4-3              
##  [63] gridExtra_2.3               rpart_4.1-15               
##  [65] stringi_1.5.3               highr_0.8                  
##  [67] S4Vectors_0.28.1            desc_1.2.0                 
##  [69] randomForest_4.6-14         foreach_1.5.1              
##  [71] e1071_1.7-4                 BiocGenerics_0.36.0        
##  [73] lava_1.6.8.1                GenomeInfoDb_1.26.2        
##  [75] rlang_0.4.10                pkgconfig_2.0.3            
##  [77] systemfonts_1.0.1           matrixStats_0.58.0         
##  [79] bitops_1.0-6                evaluate_0.14              
##  [81] purrr_0.3.4                 tensor_1.5                 
##  [83] recipes_0.1.15              labeling_0.4.2             
##  [85] transport_0.12-2            tidyselect_1.1.0           
##  [87] plyr_1.8.6                  magrittr_2.0.1             
##  [89] R6_2.5.0                    IRanges_2.24.1             
##  [91] generics_0.1.0              DelayedArray_0.16.1        
##  [93] pillar_1.5.0                withr_2.4.1                
##  [95] mgcv_1.8-34                 survival_3.2-7             
##  [97] abind_1.4-5                 RCurl_1.98-1.2             
##  [99] nnet_7.3-15                 tibble_3.1.0               
## [101] crayon_1.4.1                utf8_1.1.4                 
## [103] rmarkdown_2.7               grid_4.0.4                 
## [105] data.table_1.14.0           ModelMetrics_1.2.2.2       
## [107] digest_0.6.27               textshaping_0.3.1          
## [109] stats4_4.0.4                munsell_0.5.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Hector Roux de Bezieux, Koen Van den Berge, Kelly Street.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link external-link">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
